benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/goal/explorationspace/workload/users/UserValuesYamlProtocol.scala

```diff
@@ -52,46 +53,66 @@ object ExplorationValuesIntYamlProtocol extends DefaultYamlProtocol {
           },
           keyString(ValuesKey))
 
-      } yield IntValues(values = values)
+      } yield UserValues(values = values)
 
     }
 
-    override def write(obj: Try[IntValues]): YamlValue = unsupportedWriteOperation
+    override def write(obj: Try[UserValues]): YamlValue = unsupportedWriteOperation
   }
 
-  implicit object ExplorationSpaceUsersWriteFormat extends YamlFormat[IntValues] {
+  implicit object UserValuesWriteFormat extends YamlFormat[UserValues] {
 
-    override def write(obj: IntValues): YamlValue = YamlObject {
+    override def write(obj: UserValues): YamlValue = YamlObject {
 
       Map[YamlValue, YamlValue](
         ValuesKey -> obj.values.toYaml)
 
     }
 
-    override def read(yaml: YamlValue): IntValues = unsupportedReadOperation
+    override def read(yaml: YamlValue): UserValues = unsupportedReadOperation
+
+  }
+
+  private def assertValidValues(values: List[Int]): Try[List[Int]] = {
+
+    // check that all values are in increasing order
+    val increasingOrder = values.sliding(2).forall {
+      case List(x, y) => x < y
+    }
+
+    if (!increasingOrder) {
+      Failure(new BenchFlowDeserializationException("values must be increasing", new Throwable))
+    } else if (values.head <= 0) {
+      Failure(new BenchFlowDeserializationException("values must be > 0", new Throwable))
+    } else {
+      Success(values)
+    }
 
   }
 
   private def getRangeList(range: List[Int]): Try[List[Int]] = {
 
     if (range.size != 2) {
```

Comment:
It is similar but not exactly the same. I wanted it to be explicit therefore I did not put it into a common method.